packages/app-aco/src/components/AdvancedSearch/QueryBuilder/QueryBuilder.tsx

```diff
@@ -0,0 +1,28 @@
+import React, { useEffect, useState } from "react";
+
+import { QueryBuilder as QueryBuilderComponent } from "./components/QueryBuilder";
+import { QueryBuilderPresenter } from "./adapters/QueryBuilderPresenter";
+import { FieldRaw } from "~/components/AdvancedSearch/QueryBuilder/domain";
+import { FormAPI } from "@webiny/form";
+
+interface QueryBuilderProps {
+    fields: FieldRaw[];
+    onForm: (form: FormAPI) => void;
+    onSubmit: (data: any) => void;
+}
+
+export const QueryBuilder = ({ fields, onForm, onSubmit }: QueryBuilderProps) => {
+    const [presenter, setPresenter] = useState<QueryBuilderPresenter | undefined>();
+
+    useEffect(() => {
+        if (fields) {
```

Comment:
Is it at all possible that `fields` are undefined? Because the Props type tells me `fields` will always exist. If they do exist always, then you can remove this `useEffect` completely, and just instantiate the presenter right there in the useState, like this:

```jsx
const [presenter] = useState(() => new QueryBuilderPresenter(fields));
```